activerecord/lib/active_record/reflection.rb

```diff
@@ -708,7 +708,7 @@ def automatic_inverse_of
                 plural_inverse_name = ActiveSupport::Inflector.pluralize(inverse_name)
                 reflection = klass._reflect_on_association(plural_inverse_name)
 
-                if reflection && !active_record.automatically_invert_plural_associations
+                if valid_inverse_reflection?(reflection) && !active_record.automatically_invert_plural_associations
```

Comment:
`valid_inverse_reflection` calls `.name` on it's argument, but given this branch had a truthiness check, I assume that could blow up if `reflection` is falsey.

We should probably either keep the truthiness check first, or use `&.name` in `valid_inverse_reflection?`.

It would also be good to have test coverage for this, to be sure it behaves as expected.

Reply:
> `valid_inverse_reflection` calls `.name` on its argument, but given this branch had a truthiness check, I assume that could blow up if reflection is falsey.
> 
> We should probably either keep the truthiness check first, or use `&.name` in `valid_inverse_reflection?`.


`valid_inverse_reflection?` has a truthiness check first thing: https://github.com/rails/rails/blob/8e46af8c9396e731aaf5269182138c22621a7aa0/activerecord/lib/active_record/reflection.rb#L741-L742

That's why I just replaced the existing truthiness check, as I mentioned in the PR description. 

> It would also be good to have test coverage for this, to be sure it behaves as expected.

The [original PR](https://github.com/rails/rails/pull/50883) that introduced this flag and deprecation warning didn't add any tests, and this is a fairly tiny change over it 🤔